File: messaging/fcm-topics-codelab/starter/StockNewsApp/app/src/main/java/com/ticker/stocknews/MainActivity.java
package com.ticker.stocknews;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.LifecycleOwner;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

/**
 * Main Activity for StockNews application
 */
public class MainActivity extends AppCompatActivity implements LifecycleOwner {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    RecyclerView recyclerView = findViewById(R.id.rv_main);
    SubscriptionAdapter subscriptionAdapter = new SubscriptionAdapter();
    recyclerView.setAdapter(subscriptionAdapter);

    DividerItemDecoration dividerItemDecoration =
        new DividerItemDecoration(MainActivity.this, DividerItemDecoration.VERTICAL);
    recyclerView.addItemDecoration(dividerItemDecoration);
  }
}

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainActivity.kt
package com.google.samples.quickstart.admobexample.kotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import com.google.samples.quickstart.admobexample.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findNavController(R.id.nav_host_fragment).setGraph(R.navigation.nav_graph_kotlin)
    }
}
